app/smc/pytest/e2e_smoke.py

```diff
@@ -249,6 +265,69 @@ def upgrade_from_version_test(
     ) == read_telem(0, TAG_CM_FW_VERSION)
 
 
+def pvt_comprehensive_test(arc_chip_dut, asic_id):
+    fail_count = 0
+    arc_chip = pyluwen.detect_chips()[asic_id]
+    test_sensors = [
+        (TT_SMC_MSG_READ_TS, 0),
+        (TT_SMC_MSG_READ_PD, 19),
+        (TT_SMC_MSG_READ_VM, 0),
+    ]
+
+    for msg_type, sensor_param in test_sensors:
+        response = arc_chip.arc_msg(msg_type, True, False, sensor_param, 0, 5000)
+
+        if response[0] == 0 or response[1] != 0:
+            fail_count += 1
+    return fail_count
+
+
+def voltage_monitors_test(arc_chip_dut, asic_id):
+    arc_chip = pyluwen.detect_chips()[asic_id]
+    fail_count = 0
+
+    for sensor_id in range(NUM_VM):
+        response = arc_chip.arc_msg(TT_SMC_MSG_READ_VM, True, False, sensor_id, 0, 5000)
+
+        voltage = convert_telemetry_to_float(response[0])
+        if voltage < 0 or voltage > 1 or response[1] != 0:
+            fail_count += 1
+
+    return fail_count
+
+
+def process_detectors_test(arc_chip_dut, asic_id):
+    arc_chip = pyluwen.detect_chips()[asic_id]
+    fail_count = 0
+
+    delay_chains = [19, 20, 21]
+    for delay_chain in delay_chains:
+        for sensor_id in range(NUM_PD):
+            response = arc_chip.arc_msg(
+                TT_SMC_MSG_READ_PD, True, False, delay_chain, sensor_id, 5000
+            )
+
+            freq = convert_telemetry_to_float(response[0])
+            if freq < 100 or freq > 240 or response[1] != 0:
+                fail_count += 1
+
+    return fail_count
+
+
+def temperature_sensors_test(arc_chip_dut, asic_id):
+    arc_chip = pyluwen.detect_chips()[asic_id]
+    fail_count = 0
+
+    for sensor_id in range(NUM_TS):
+        response = arc_chip.arc_msg(TT_SMC_MSG_READ_TS, True, False, sensor_id, 0, 5000)
+
+        temp = convert_telemetry_to_float(response[0])
+        if temp < 40 or temp > 70 or response[1] != 0:
+            fail_count += 1
```

Comment:
Please add logging for failure cases (i.e. what failed, expected vs actual values, what iteration)